### PR TITLE
Switch to GPU enabled base image

### PR DESCRIPTION
**Pull request by sweatybridge**
- `nvidia/cuda:10.0-cudnn7-runtime-ubuntu18.04` is the latest supported by GCP
- `torchvision==0.5.0` and `torch==1.4.0` are versions required by cuda 10

Tested on GCP
- Had to remove `--preload` flag to avoid worker being restarted on every request

### CODE REVIEW

File: flask-gunicorn/entrypoint.sh [hunk 1]
```diff
@@ -24,6 +24,5 @@ exec gunicorn serve_http:app \
     "$EXTRA_OPTS" \
     --bind=":${BEDROCK_SERVER_PORT:-8080}" \
     --worker-class=gthread \
-    --workers="$WORKERS" \
```
Comment (nglinh) [inline flask-gunicorn/entrypoint.sh:27]: is change from worker to threads intentional?

**Review by nglinh — COMMENTED** [review]
Any update needed for fastapi-gunicorn too?

File: flask-gunicorn/entrypoint.sh (hunk 1 above)
Comment (sweatybridge) [reply flask-gunicorn/entrypoint.sh:27]: Yes, there's problem with multiprocess talking to cuda driver.

**Comment by sweatybridge** [comment]
@nglinh No plan to support fastapi yet. Our internal benchmark shows fastapi to be slower and buggy compared to flask.

**Comment by nglinh** [comment]
My only concern is with having cuda baked in to the main dockerfile since all our subsequent images will have cuda base. How about splitting it to a different docker file so we can keep cuda and non-cuda image separated

**Comment by sweatybridge** [comment]
Yea, let's have a separate dockerfile for gpu support. It should support most, if not all, of the build args of the current dockerfile.

**Comment by sweatybridge** [comment]
While you are at it, also think about if it makes sense to put all cuda related files into a separate directory, like flask-cuda. It feels a bit cleaner than switching on some env var or build arg.

**Pull request merged**